runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7642,7 +7655,20 @@ done:;
 		_sp += 1;
 		if (NULL == obj) {
 			rc = THROW_NPE;
-		} else {
+		} else
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		if (J9_IS_J9CLASS_VALUETYPE(J9OBJECT_CLAZZ(_currentThread, obj))) {
+			J9Class * objClass = J9OBJECT_CLAZZ(_currentThread, obj);
```

Comment:
double fetch of the class

Reply:
If we don't double fetch the class, we might have to use a goto statement here. Is it fine?

Reply:
I think GAC just means that the class should be cached instead of fetched twice. 

I proposed this approach because it would allow us to keep the non-VT code unchanged. Also the double fetch is on the exception path, so even if the compile doesn't cache it for us, it is a slow path regardless.

That being said we could re-write as:

```
if (NULL == obj) {
   ...
} else {
ifdef VT
   Class clazz = //fetch and save class
   if (class.isVT) {
      ...
   } else
endif VT
  {
      //rest of code
  }
}
```